lib/sparse/src/index/mod.rs

```diff
@@ -0,0 +1,6 @@
+#[allow(dead_code)]
+mod inverted_index;
+#[allow(dead_code)]
+mod posting_list;
+#[allow(dead_code)]
+mod search_context;
```

Comment:
I think this should also work? 🤔

```suggestion
#![allow(dead_code)]

mod inverted_index;
mod posting_list;
mod search_context;
```

Or add `#![allow(dead_code)]` _inside_ each module. Looks better this way, IMO.